mixins/arrow-keys-mixin.js

```diff
@@ -0,0 +1,141 @@
+import { html } from 'lit-element/lit-element.js';
+
+const keyCodes = Object.freeze({
+	END: 35,
+	HOME: 36,
+	LEFT: 37,
+	UP: 38,
+	RIGHT: 39,
+	DOWN: 40,
+});
+
+export const ArrowKeysMixin = superclass => class extends superclass {
+	static get properties() {
+		return {
+			arrowKeysDirection: { type: String },
+			arrowKeysNoWrap: { type: Boolean }
+		};
+	}
+
+	constructor() {
+		super();
+		this.arrowKeysDirection = 'leftright';
+		this.arrowKeysNoWrap = false;
+	}
+
+	arrowKeysContainer(inner) {
+		return html`<div class="arrow-keys-container" @keyup="${this._handleArrowKeys}">
+			${inner}
+		</div>`;
+	}
+
+	async arrowKeysFocusablesProvider() {
+		return [...this.shadowRoot.querySelectorAll('.d2l-arrowkeys-focusable')];
```

Comment:
Is this being done to convert to an array?  If so, it this to work around some browser inconsistency?